test/commons/matches/from-function.js

```diff
@@ -0,0 +1,88 @@
+describe('matches.fromFunction', function () {
+  var fromFunction = axe.commons.matches.fromFunction
+  function noop () {}
+
+  it('throws an error when the matcher is a number', function () {
+    assert.throws(function () {
+      fromFunction(noop, 123)
+    })
+  })
+
+  it('throws an error when the matcher is a string', function () {
+    assert.throws(function () {
+      fromFunction(noop, 'foo')
+    })
+  })
+
+  it('throws an error when the matcher is an array', function () {
+    assert.throws(function () {
+      fromFunction(noop, ['foo'])
+    })
+  })
+
+  it('throws an error when the matcher is a RegExp', function () {
+    assert.throws(function () {
+      fromFunction(noop, /foo/)
+    })
+  })
+
+  describe('with void matchers', function () {
```

Comment:
what is the thinking behind allowing null or undefined in `.fromFunction` matcher?

Reply:
Probably wasn't necessary. I'm going to remove it.